### PR TITLE
Document mirage `Ipaddr` conversion

### PR DESCRIPTION
**Pull request by RyanGibb**
Enabled by https://github.com/mirage/ocaml-ipaddr/pull/117

### CODE REVIEW

File: lib_eio/net.mli [hunk 1]
```diff
@@ -27,7 +27,12 @@ type Exn.err += E of error
 val err : error -> exn
 (** [err e] is [Eio.Exn.create (Net e)] *)
 
-(** IP addresses. *)
+(** IP addresses.
+
+  In order to interoperate with the {{:https://opam.ocaml.org/packages/ipaddr/} ipaddr} library we can use it's octet conversion functions.
```
Comment (patricoferris) [inline lib_eio/net.mli:32]:
```suggestion
  In order to interoperate with the {{:https://ocaml.org/p/ipaddr/} ipaddr} library we can use its octet conversion functions.
```

I think linking to OCaml.org might be preferable and more future proof?

File: lib_eio/net.mli [hunk 2]
```diff
@@ -27,7 +27,12 @@ type Exn.err += E of error
...
+  That is, to convert to an [Eio.Ipaddr] we can use [Ipaddr.to_octets ip |> Eio.Net.Ipaddr.of_raw].
```
Comment (patricoferris) [inline lib_eio/net.mli:33]:
```suggestion
  That is, to convert to an {! Eio.Net.Ipaddr.t} we can use [Ipaddr.to_octets ip |> Eio.Net.Ipaddr.of_raw].
```

That should link to the `t` in the docs :)) (I think)

**Review by patricoferris — COMMENTED** [review]
Thanks! This is much cleaner than what we had to do before ^^

**Review by talex5 — COMMENTED** [review]
I suggest putting it as a section with `of_raw`, e.g.

```ocaml
  (** {2 Interoperability}

      To convert to or from OCaml Unix addresses, use {!Eio_unix.Ipaddr}.

      To interoperate with the {{:https://opam.ocaml.org/packages/ipaddr/} ipaddr} library:
      - [Ipaddr.to_octets ipaddr_ip |> Eio.Net.Ipaddr.of_raw]
      - [Ipaddr.of_octets_exn (eio_ip :> string)] *)

  val of_raw : string -> v4v6
  (** [of_raw addr] casts [addr] to an IP address.
      @raise Invalid_argument if it is not 4 or 16 bytes long. *)
```

File: lib_eio/net.mli (hunk 1 above)
Comment (RyanGibb) [reply lib_eio/net.mli]: Thanks patrick, that makes sense!

File: lib_eio/net.mli (hunk 2 above)
Comment (RyanGibb) [reply lib_eio/net.mli]: Ah yes, I'm not too familiar with OCaml documentation syntax. Thanks!

**Comment by RyanGibb** [comment]
Thanks for the suggestion @talex5, I've applied it now

**Review by avsm — APPROVED** [review]

**Comment by talex5** [comment]
Thanks!

**Pull request merged**